source/exec.tex

```diff
@@ -7709,10 +7671,68 @@
 
 \rSec2[exec.scope.concepts]{Execution scope concepts}
 
+\pnum
+The \libconcept{scope_association} concept defines
+the requirements on a type \tcode{Assoc}.
+An object of type \tcode{Assoc} is \defn{engaged}
+if and only if it owns an association with an async scope,
+referred to as its \defnadj{associated}{scope}.
+
+\begin{codeblock}
+namespace std::execution {
+  template<class Assoc>
+    concept @\deflibconcept{scope_association}@ =
+      @\libconcept{movable}@<Assoc> &&
+      is_nothrow_move_constructible_v<Assoc> &&
+      is_nothrow_move_assignable_v<Assoc> &&
+      @\libconcept{default_initializable}@<Assoc> &&
+      requires(const Assoc assoc) {
+        { static_cast<bool>(assoc) } noexcept;
+        { assoc.try_associate() } -> @\libconcept{same_as}@<Assoc>;
+      };
+}
+\end{codeblock}
+
+\pnum
+A type \tcode{Assoc} models \libconcept{scope_association} only if:
+\begin{itemize}
+\item
+a default constructed object of type \tcode{Assoc} is not engaged;
+\item
+for an object \exposid{assoc} of type \tcode{Assoc},
+\tcode{static_cast<bool>(assoc)} is \tcode{true}
+if and only if \tcode{assoc} is engaged;
+\item
+no two distinct objects of type \tcode{Assoc} own the same assocation;
+\item
+for an object \tcode{assoc} of type \tcode{Assoc},
+destroying \tcode{assoc} releases the assocation owned by \tcode{assoc}, if any;
+\item
+for an object \tcode{assoc} of type \tcode{Assoc},
+after it is used as the source operand of a move constructor,
+the \tcode{assoc} is not engaged;
+\item
+for distinct objects \tcode{assoc1} and \tcode{assoc2} of type \tcode{Assoc},
+after evaluating \tcode{assoc1 = std::move(assoc2)},
+the association owned by \tcode{assoc1}, if any, is released and
+\tcode{assoc2} is not engaged;
+\item
+for an object \tcode{assoc} of type \tcode{Assoc} that is engaged,
+\tcode{assoc.try_associate()} either
+returns an object that is not engaged or
+acquires a new association with \tcode{assoc}'s associated scope and
+returns an engaged object that owns that association;
+\item
+for an object \tcode{assoc} of type \tcode{Assoc} that is not engaged,
+\tcode{assoc.try_associate()} returns an object that is not engaged.
+\end{itemize}
+
 \pnum
 The \libconcept{scope_token} concept defines the requirements on
 a type \tcode{Token} that can be used to create associations
 between senders and an async scope.
+Every object of type \tcode{Token} is associated with an async scope
+that is referred to as its \term{associated scope}.
```

Comment:
Should this be \defnadj ? It's a new definition of "associated scope" (for a scope token this time, not a scope association).